django/db/models/query.py

```diff
@@ -815,6 +815,25 @@ def complex_filter(self, filter_obj):
         else:
             return self._filter_or_exclude(None, **filter_obj)
 
+    def _combinator_query(self, combinator, *other_qs, **kwargs):
+        # Clone the query so we inherit the select list and everything
+        clone = self._clone()
+        # Clear limits and ordering, so they can be reapplied
+        clone.query.clear_ordering(True)
+        clone.query.clear_limits()
+        clone.query.combined_queries = (self.query,) + tuple(qs.query for qs in other_qs)
+        clone.query.combinator = (combinator, kwargs.pop('all', False))
```

Comment:
I'm not sure I like using a tuple here instead of two separate attributes. There's a lot of `combinator[0]` and the occasional `combinator[1]` that is not immediately obvious to the reader. Perhaps `combinator_all` is a suitable name for the second part?